src/dmd/root/ctfloat.d

```diff
@@ -229,4 +229,13 @@ extern (C++) struct CTFloat
         minusone = real_t(-1);
         half = real_t(0.5);
     }
+
+    /// Deinitalize the compile-time floating-point globals.
+    @trusted
+    static void deinitialize()
+    {
+        import dmd.root.utilities : resetStaticFields;
+
+        resetStaticFields!(typeof(this));
```

Comment:
1. There don't appear to be any static fields in CTfloat needing to be reset.
2. There's no reason to use `typeof(this)` instead of simply `CTfloat`
3. Because of (1), which is the only user of resetStaticFields, and which has no fields to reset, there appears to be no purpose to adding dmd.root.utilities.

Reply:
> There don't appear to be any static fields in CTfloat needing to be reset.

`__gshared` fields are implicitly `static`. These fields need to be reset:

https://github.com/dlang/dmd/blob/8d760396c9a5e663614cc81175806f89eb0b6225/src/dmd/root/ctfloat.d#L219-L222

> There's no reason to use typeof(this) instead of simply CTfloat

I can change that.

> Because of (1), which is the only user of resetStaticFields, and which has no fields to reset, there appears to be no purpose to adding dmd.root.utilities.

There are more fields in other types that need to be reset, like in `dmd.mtype.Type`.

Reply:
> These fields need to be reset:

I don't see why. Then they'll be initialized again to exactly the same values.

Reply:
I removed `resetStaticFields` now anyway.

Reply:
> like in dmd.mtype.Type

Initializing fields should be done in the initializer code, not the de-initializer code. De-initialization implies release of resources, and this code does not (and cannot) release any resources.